internal/cli/atlas_alert_config_delete.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mcli/internal/flags"
+	"github.com/mongodb/mcli/internal/store"
+	"github.com/mongodb/mcli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type atlasAlertConfigDeleteOpts struct {
+	*globalOpts
+	entry          string
+	successMessage string
+	failMessage    string
+	store          store.AlertConfigurationDeleter
+}
+
+func (opts *atlasAlertConfigDeleteOpts) init() error {
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *atlasAlertConfigDeleteOpts) Run() error {
+	error := opts.store.DeleteAlertConfiguration(opts.ProjectID(), opts.entry)
```

Comment:
since deleting is final we usually confirm with users before deleting, since this is quite repetitive code I encapsulate some of the logic in `deleteOpts`, you can then just do 
```go
func (opts *atlasWhitelistDeleteOpts) Run() error {
	return opts.DeleteFromProject(opts.store. DeleteAlertConfiguration, opts.ProjectID())
}
``` 
And `DeleteFromProject` will take care of confirming and printing the messages